apps/client/src/features/app-settings/panel/project-panel/ProjectListItem.tsx

```diff
@@ -0,0 +1,176 @@
+  createdAt,
+  editingFilename,
+  editingMode,
+  filename,
+  onSubmit,
+  onToggleEditMode,
+  updatedAt,
+}: ProjectListItemProps) {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+
+  const handleRefetch = async () => {
+    await ontimeQueryClient.invalidateQueries({ queryKey: PROJECT_LIST });
+  };
+
+  const handleSubmitRename = useCallback(
+    async (values: DuplicateRenameProjectFormValues) => {
+      try {
+        setSubmitError(null);
+
+        if (!values.filename) {
+          setSubmitError('Filename cannot be blank');
+          return;
+        }
+        await renameProject(filename, values.filename);
+        await handleRefetch();
+        onSubmit();
+      } catch (error) {
+        if (error instanceof AxiosError) {
+          const errorMessage = error?.response?.data?.message;
+
+          setSubmitError(errorMessage);
+        } else {
+          setSubmitError('An unknown error occurred');
+        }
+      }
+    },
+    [filename, onSubmit],
+  );
+
+  const handleSubmitDuplicate = useCallback(
+    async (values: DuplicateRenameProjectFormValues) => {
+      try {
+        setSubmitError(null);
+
+        if (!values.filename) {
+          setSubmitError('Filename cannot be blank');
+          return;
+        }
+        await duplicateProject(filename, values.filename);
+        await handleRefetch();
+        onSubmit();
+      } catch (error) {
+        if (error instanceof AxiosError) {
+          const errorMessage = error?.response?.data?.message;
+
+          setSubmitError(errorMessage);
+        } else {
+          setSubmitError('An unknown error occurred');
+        }
+      }
+    },
+    [filename, onSubmit],
+  );
+
+  const handleToggleEditMode = useCallback(
+    (editMode: EditMode, filename: string | null) => {
+      setSubmitError(null);
+      onToggleEditMode(editMode, filename);
+    },
+    [onToggleEditMode],
+  );
+
+  const handleCancel = useCallback(() => {
+    handleToggleEditMode(null, null);
+  }, [handleToggleEditMode]);
+
+  const isCurrentlyBeingEdited = editingMode && filename === editingFilename;
+
+  return (
+    <tr key={filename} className={current ? style.current : undefined}>
+      <td colSpan={isCurrentlyBeingEdited ? 99 : 1}>
+        {isCurrentlyBeingEdited ? (
+          <DuplicateRenameProjectForm
+            action={editingMode}
+            filename={filename}
+            onSubmit={editingMode === 'duplicate' ? handleSubmitDuplicate : handleSubmitRename}
+            onCancel={handleCancel}
+            submitError={submitError}
+          />
+        ) : (
+          <span>{filename}</span>
+        )}
+      </td>
+      {!isCurrentlyBeingEdited && (
+        <>
+          <td>{createdAt}</td>
+          <td>{updatedAt}</td>
+          <td className={style.actionButton}>
+            <ActionMenu current={current} filename={filename} onChangeEditMode={handleToggleEditMode} />
+          </td>
+        </>
+      )}
+    </tr>
+  );
```

Comment:
suggestion: how about this? maybe it is easier to read?

```suggestion
  return (
    <tr key={filename} className={current ? style.current : undefined}>
      {isCurrentlyBeingEdited ? (
        <td colSpan={99}>
          <DuplicateRenameProjectForm
            action={editingMode}
            filename={filename}
            onSubmit={editingMode === 'duplicate' ? handleSubmitDuplicate : handleSubmitRename}
            onCancel={handleCancel}
            submitError={submitError}
          />
        </td>
      ) : (
        <>
          <td>{filename}</td>
          <td>{createdAt}</td>
          <td>{updatedAt}</td>
          <td className={style.actionButton}>
            <ActionMenu current={current} filename={filename} onChangeEditMode={handleToggleEditMode} />
          </td>
        </>
      )}
    </tr>
  );
```

Reply:
it does, nice